src/app/components/meme/meme.component.ts

```diff
@@ -4,17 +4,26 @@ import { Component, Input, Output, EventEmitter } from '@angular/core';
 @Component({
   selector: 'app-meme',
   templateUrl: './meme.component.html',
-  styleUrls: ['./meme.component.scss']
+  styleUrls: ['./meme.component.scss'],
 })
 export class MemeComponent {
-
   @Input()
   meme: Meme;
 
   @Input()
   isUpvoted: boolean;
 
+  @Input()
+  isAdmin: boolean;
+
   @Output()
   like = new EventEmitter<void>();
 
+  @Output()
+  approvedChange = new EventEmitter();
+
+  changeApproval() {
+    const newMeme = { ...this.meme, approved: !this.meme.approved };
+    this.approvedChange.emit(newMeme);
```

Comment:
I love how you keep this component dumb and stateless. It doesn't touch the state, just communicates the action up to the parent. Good one!